bundles/org.openhab.binding.souliss/src/main/java/org/openhab/binding/souliss/internal/handler/SoulissGatewayHandler.java

```diff
@@ -0,0 +1,324 @@
+                        .intValue();
+                logger.debug("Get ping refresh interval: {}", pingRefreshInterval);
+            }
+            if (gwConfigurationMap.get(SoulissBindingConstants.CONFIG_SUBSCRIPTION_REFRESH) != null) {
+                subscriptionRefreshInterval = ((BigDecimal) gwConfigurationMap
+                        .get(SoulissBindingConstants.CONFIG_SUBSCRIPTION_REFRESH)).intValue();
+                logger.debug("Get ping refresh interval: {}", pingRefreshInterval);
+            }
+            if (gwConfigurationMap.get(SoulissBindingConstants.CONFIG_HEALTHY_REFRESH) != null) {
+                healthRefreshInterval = ((BigDecimal) gwConfigurationMap
+                        .get(SoulissBindingConstants.CONFIG_HEALTHY_REFRESH)).intValue();
+                logger.debug("Get health refresh interval: {}", healthRefreshInterval);
+            }
+
+            if (gwConfigurationMap.get(SoulissBindingConstants.CONFIG_SEND_REFRESH) != null) {
+                sendRefreshInterval = ((BigDecimal) gwConfigurationMap.get(SoulissBindingConstants.CONFIG_SEND_REFRESH))
+                        .intValue();
+                logger.debug("Get send refresh interval: {}", sendRefreshInterval);
+            }
+
+            if (gwConfigurationMap.get(SoulissBindingConstants.CONFIG_TIMEOUT_TO_REQUEUE) != null) {
+                sendTimeoutToRequeue = ((BigDecimal) gwConfigurationMap
+                        .get(SoulissBindingConstants.CONFIG_TIMEOUT_TO_REQUEUE)).intValue();
+                logger.debug("Get send timeout to requeue: {}", sendTimeoutToRequeue);
+            }
+
+            if (gwConfigurationMap.get(SoulissBindingConstants.CONFIG_TIMEOUT_TO_REMOVE_PACKET) != null) {
+                sendTimeoutToRemovePacket = ((BigDecimal) gwConfigurationMap
+                        .get(SoulissBindingConstants.CONFIG_TIMEOUT_TO_REMOVE_PACKET)).intValue();
+                logger.debug("Get send timeout to requeue: {}", sendTimeoutToRequeue);
+            }
+
+            // START SERVER ON DEFAULT PORT - Used for topics
+            if (udpServerDefaultPortRunnableClass == null) {
+                logger.debug("Starting UDP server on Souliss Default Port for Topics (Publish&Subcribe)");
+                datagramSocketDefaultPort = SoulissDatagramSocketFactory.getSocketDatagram(soulissGatewayPort,
+                        this.logger);
+                if (datagramSocketDefaultPort != null) {
+                    udpServerDefaultPortRunnableClass = new SoulissBindingUDPServerJob(datagramSocketDefaultPort,
+                            SoulissBindingNetworkParameters.discoverResult);
+                    // Changes from scheduleAtFixedRate - Luca Calcaterra
+                    scheduler.scheduleWithFixedDelay(udpServerDefaultPortRunnableClass, 100,
+                            SoulissBindingConstants.SERVER_CICLE_IN_MILLIS, TimeUnit.MILLISECONDS);
```

Comment:
If I see correctly the executed task is blocking. This will block the framework's thread pool. You could execute this in a new thread.